beacon-chain/simulator/service.go

```diff
@@ -155,6 +154,10 @@ func (sim *Simulator) run(slotInterval <-chan uint64, requestChan <-chan p2p.Mes
 				powChainRef = []byte{byte(slot)}
 			}
 
+			slotsStart := cState.LastStateRecalculationSlot() - params.GetConfig().CycleLength
+			slotIndex := (slot - 1 - slotsStart) % params.GetConfig().CycleLength
+			shardID := cState.ShardAndCommitteesForSlots()[slotIndex].ArrayShardAndCommittee[0].Shard
```

Comment:
btw, this is a good example of a method that belongs in `types/crystallized_state.go`, so that this can be simplified to `cState.GetShardsAndCommitteesForSlot(slot).ArrayShardAndCommittee[0].Shard`.

Reply:
Awesome. Added a method in `types/crystallized_state.go`